src/SlackLogger.php

```diff
@@ -13,113 +14,171 @@
-	{
-		$ctxOptions = [
-			'http' => [
-				'method' => 'POST',
-				'header' => 'Content-type: application/x-www-form-urlencoded',
-				'content' => http_build_query($postContent),
-			],
-		];
-		$ctx = stream_context_create($ctxOptions);
-		$resultStr = @file_get_contents($url, NULL, $ctx);
-
-		if ($resultStr != 'ok') {
-			throw new \RuntimeException('Error sending request to the Slack API: ' . $http_response_header[0]);
-		}
-	}
+    /** @var string */
+    private $slackUrl;
+    /** @var string */
+    private $logUrl;
+    /** @var string */
+    private $channel;
+    /** @var string */
+    private $username;
+    /** @var string */
+    private $icon;
+    /** @var string */
+    private $pretext;
+    /** @var int|string The minimum interval between two events. */
+    private $interval = '1 day';
+    /** @var string|null File where the filter persists the state. */
+    private $file;
+    /** @var bool Indicates whether problems with the file write-ability was already logged (prevents recursion). */
+    private $loggedUnwriteableFile = false;
+
+
+    public function __construct($slackUrl, $logUrl, $channel = null, $username = null, $icon = null, $pretext = null, $file = null, $interval = null)
+    {
+        parent::__construct(Debugger::$logDirectory, Debugger::$email, Debugger::getBlueScreen());
+        $this->slackUrl = $slackUrl;
+        $this->logUrl = $logUrl;
+        $this->channel = $channel;
+        $this->username = $username;
+        $this->icon = $icon;
+        $this->pretext = $pretext;
+        $this->file = $file;
+        $this->interval = $interval;
+    }
+
+
+    /**
+     * @inheritdoc
+     */
+    public function log($value, $priority = self::INFO)
+    {
+        $logFile = parent::log($value, $priority);
+
+        $message = ucfirst($priority) . ': ';
+        if ($value instanceof Exception || $value instanceof \Throwable) { // NOTE: backwards compatibility with PHP 5
+            $message .= $value->getMessage();
+        } elseif (is_array($value)) {
+            $message .= reset($value);
+        } else {
+            $message .= (string)$value;
+        }
+
+        if ($logFile && $this->logUrl) {
+            $message .= ' (<' . str_replace('__FILE__', basename($logFile), $this->logUrl) . '|Open log file>)';
+        }
+
+        $notify = $this->isAllowedByInterval();
+        //add logged message to buffer
+        $logSuccessful = (bool) @file_put_contents($this->getFile(), "\n" . date("r") . "\n" . $message, FILE_APPEND);
+
+        if (!$logSuccessful && !$this->loggedUnwriteableFile) {
+            $this->loggedUnwriteableFile = true;
+            trigger_error("Unable to write to file '{$this->getFile()}'. Filter will deny the incoming events.", E_USER_WARNING);
+        }
+
+        if ($notify) {
+            //if interval after last update has passed, flush all messages in queue to Slack
+            $this->sendSlackMessage(@file_get_contents($this->getFile()), $priority);
```

Comment:
Please, do not archive all the messages. Use the file as a mere flag, telling when the last message was really notified. Swallow any messages not allowed by the interval.
In case the interval is actually set (i.e., the interval check is active), append a short explanation that the following X will not be notified (where X is the human-readable description of the interval). Please note this all must fit in the overall message length limit.